sources/ElkArte/Graphics/Gd2.php

```diff
@@ -0,0 +1,494 @@
+<?php
+
+/**
+ * This file deals with low-level graphics operations performed on images,
+ * specially as needed for the GD library
+ *
+ * @package   ElkArte Forum
+ * @copyright ElkArte Forum contributors
+ * @license   BSD http://opensource.org/licenses/BSD-3-Clause (see accompanying LICENSE.txt file)
+ *
+ * @version 2.0 dev
+ *
+ */
+
+namespace ElkArte\Graphics;
+
+/**
+ * Class Gd2
+ *
+ * @package ElkArte\Graphics
+ */
+class Gd2 extends AbstractManipulator
+{
+	/** @var resource */
+	protected $_image;
```

Comment:
This should be declared in the AbstractManipulator class, otherwise scrutinizer complains that here https://github.com/elkarte/Elkarte/pull/3280/files#diff-d62ebe3b9dd825fc67c0d66effaeba72R119 the property ->_image is not defined. :D

Reply:
I'll add it there and see how it goes ... I moved it to the specific manipulators since $_image can be either a gd resource or Imagick one.   If you only declare it in the abstract as both (\Imagick|resource) then Storm lights up with unknown method errors, so I'm going to leave them in the manipulators as well.